pkg/build/controller/build/build_controller.go

```diff
@@ -384,59 +421,6 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 	return podSpec, nil
 }
 
-// resolveOutputDockerImageReference returns a Docker image name
-// computed from the build.Spec.Output.To reference.
-// Note that we are using controller level permissions to resolve the imagestream and secret,
-// meaning users could theoretically define a build that references an imagestream they cannot
-// see, and 1) get the docker image reference of that imagestream and 2) a reference to a secret
-// associated with a service account that can push to that location.  However they still cannot view the secret,
-// and ability to use a service account implies access to its secrets, so this is considered safe.
-// Furthermore it's necessary to enable triggered builds since a triggered build is not "requested"
-// by a particular user, so there are no user permissions to validate against in that case.
```

Comment:
@deads2k asked very nicely for all this commenting and you've removed it in this PR... :(